core/src/ops/quant.rs

```diff
@@ -146,6 +147,119 @@ fn info_quantize_linear_i8(q: &QuantizeLinearI8) -> TractResult<Vec<String>> {
+        .max(u8::min_value() as i32)
+        .min(u8::max_value() as i32) as u8
+}
+
+fn dynamic_quantize_linear_u8(
+    scale: f32,
+    zero_point: u8,
+    xs: &[f32],
+    ys: &mut [u8],
+) -> TractResult<()> {
+    xs.iter()
+        .zip(ys.iter_mut())
+        .for_each(|(x, y)| *y = dynamic_quantize_linear_f32_u8(*x, scale, zero_point));
+    Ok(())
+}
+
+#[derive(Clone, Debug, new, Educe)]
+#[educe(Hash)]
+pub struct DynamicQuantizeLinearU8;
+
+impl Op for DynamicQuantizeLinearU8 {
+    fn name(&self) -> Cow<str> {
+        "DynamicQuantizeLinearU8".into()
+    }
+
+    fn info(&self) -> TractResult<Vec<String>> {
+        Ok(vec![])
+    }
+
+    fn validation(&self) -> Validation {
+        Validation::Accurate
+    }
+
+    op_core_mir!();
+    op_as_typed_op!();
+}
+
+tract_linalg::impl_dyn_hash!(DynamicQuantizeLinearU8);
+
+impl EvalOp for DynamicQuantizeLinearU8 {
+    fn is_stateless(&self) -> bool {
+        true
+    }
+    fn eval(&self, inputs: TVec<Arc<Tensor>>) -> TractResult<TVec<Arc<Tensor>>> {
+        let input = &inputs[0];
+        let a_input = input.to_array_view::<f32>()?;
+        let (scale, zero_point) = scale_and_zero_point(a_input);
+
+        let mut dst = unsafe { Tensor::uninitialized_dt(u8::datum_type(), input.shape())? };
+        // We cannot use quantize_linear_u8 here because it does `x * scale.recip()`
+        // instead of `x / scale`. This change some number enough to be rounded to another integer.
+        dynamic_quantize_linear_u8(
+            scale,
+            zero_point,
+            input.as_slice::<f32>()?,
+            dst.as_slice_mut::<u8>()?,
+        )?;
+
+        let quantized_tensor = dst.into_arc_tensor();
+        let scale_tensor = rctensor0(scale);
+        let zero_point_tensor = rctensor0(zero_point);
+
+        Ok(tvec!(quantized_tensor, scale_tensor, zero_point_tensor))
+    }
+}
+
+impl TypedOp for DynamicQuantizeLinearU8 {
+    fn output_facts(&self, inputs: &[&TypedFact]) -> TractResult<TVec<TypedFact>> {
+        let mut quantized_fact = inputs[0].clone();
+        quantized_fact.datum_type = u8::datum_type();
+        let shape = ShapeFact::from_dims(&[])?;
+        let scale_fact = TypedFact::dt_shape(f32::datum_type(), shape.clone())?;
+        let zero_fact = TypedFact::dt_shape(u8::datum_type(), shape)?;
+        Ok(tvec!(quantized_fact, scale_fact, zero_fact))
+    }
+
+    as_op!();
+}
+
+fn scale_and_zero_point<'a>(v: ArrayViewD<'a, f32>) -> (f32, u8) {
+    let (v_min, v_max) = v.fold((f32::MAX, f32::MIN), |(a_min, a_max), &v| {
+        if v < a_min {
+            (v, a_max)
+        } else if v > a_max {
+            (a_min, v)
+        } else {
+            (a_min, a_max)
+        }
+    });
+
+    // adjust data range to include 0
+    let min = v_min.min(0.);
+    let max = v_max.max(0.);
+
+    // quantize range
+    let min_t = 0.;
+    let max_t = 255.0;
+
+    let scale = (max - min) / max_t;
+
+    let zero_point = -min / scale;
```

Comment:
this will panic if the input array contains zeros everywhere, i.e. `max == min`. this is quite unlikely in the real world, but could easily happen in test cases. we should probably catch this case with
```suggestion
    let (scale, zero_point) = if max == min {
        (1., 0.)
    } else {
        let scale = (max - min) / max_t;
        let zero_point = -min / scale;
        (scale, zero_point)
    };
```
for the edge case, scale can of course take any value except zero, but one is always a good choice for numerical stability when it is used in the division in `dynamic_quantize_linear_f32_u8`.

and we should probably also add a test for that edge case.

Reply:
The division is between `f32` and that is well defined also when `scale` is 0. The method `round` is also well defined when the value is `inf` or `Nan`.

Reply:
for my records, it is `NaN as u8` which evaluates to `0` which makes this work.